doc/userdoc/styleguide/styleguide.rst

```diff
@@ -0,0 +1,566 @@
+"note", "tip", "warning", "admonition"
+
+If you want a custom admonition use
+
+::
+
+   .. admonition:: custom name
+
+         Here is some text
+
+Rendered as
+
+
+   .. admonition:: custom name
+
+         Here is some text
+
+
+Roles
+~~~~~
+
+See `the roles section in the Sphinx docs <https://www.sphinx-doc.org/en/
+master/usage/restructuredtext/roles.html>`_ for details.
+
+In-text math
+^^^^^^^^^^^^
+
+For in-text math, use
+
+::
+
+   Now we can see :math:`x=1` for this example.
+
+This will be rendered as
+
+
+   Now we can see :math:`x=1` for this example.
+
+Referencing
+^^^^^^^^^^^
+
+For referencing reStructuredText files in repository, use
+
+::
+
+   :doc:`file`
+
+::
+
+   :doc:`custom name <path/file>`
+
+This will be rendered as
+
+   :doc:`file`
+
+   :doc:`custom name <file>`
+
+
+For section headings, you can use either of the two examples below. But to
+reference figures or arbitrary places in a file, you must include a custom
+name (second example) in the reference for it to work.
+
+::
+
+    :ref:`ref_name`
+
+or
+
+::
+
+    :ref:`custom name <ref_name>`
+
+Rendered as
+
+
+    :ref:`ref_name`
+
+    :ref:`custom name <ref_name>`
+
+
+.. note::
+
+    The `ref_name` needs to be above the section you want to reference with the
+    following syntax
+
+    ::
+
+        .. _ref_name:
+
+        secton_header
+        --------------
+
+Special links
+^^^^^^^^^^^^^
+
+.. attention::
+
+  The items in this section are still in development and have not been incorporated into nest:master!
+
+
+To link PyNEST API functions used in-text to the API reference page use the following syntax:
+
+::
+
+   :py:func:`.Create`
+
+
+Rendered as
+
+   :py:func:`.Create`
+
+
+To link terms to the glossary page use the HoverXTooltip role from Mahdi Enan (INM-6)
+
+::
+
+  :hxt_ref:`E_L`
+
+To link to external projects (PyNN, Elephant, nestml), you can treat references as you would in your local project
+with the addition of the intersphinx unique identifer (see also conf.py >  intersphinx_mapping).
+
+::
+
+  :doc:`tuturial for nestml <nestml:tutorials>`
+  :py:func:`pyNN.utility.get_simulator`
+
+Bibliography style
+------------------
+
+The reStructuredText reference style is used throughout documentation so links
+are autogenerated and a consistent format is used.
+
+For in-text citations, we use the reStructuredText numeric style ``[1]_``.
+
+
+For example:
+
+.. code-block:: none
+
+    The following example is based on Smith [1]_.
+    [2]_ contains a technically detailed description.
+
+Please ensure your reference follows the following guidelines:
+
+* References with more than 5 authors use 'et al.'.
+* Use initials for first name of authors
+* Surname precedes first name for all authors
+* No comma follows surname
+* Full stop after every section of bibliography.
+* No formatting such as italics, bold, or underline.
+* Full title of journal
+* Article titles written in sentence case
+* Year follows author(s), in parentheses
```

Comment:
Move this after information on author names, so it is in the order of appearance in the bib entry.

Reply:
done